### PR TITLE
Restructure/reorder configuration dashboard

### LINKED ISSUE #4040: Reorder items in Configuration panel
**Home Assistant release with the issue:**
0.100.2

**Description of problem:**
I think Configuration panel could use some minor re-arranging to surface the more frequently used sections higher in the list.

Basically this boils down to moving "Users" down near the bottom and "Automation" and "Script" near the top. Here is a before (left) and after (right) screenshot, with the reasoning below.

config-panel-arrangement

**Reasoning**:
Users is currently placed near the top, yet it's probably the _least_ visited section if you think about it. An owner/admin account is created during onboarding and many users will never need to create another one. Even those who do create user accounts for each family member, it's a one-time process. Not something that you are doing regularly like the other sections here. So it should be demoted a bit.

Automation and Script Editors are probably the most visited sections inside this panel. Yet these are listed towards the bottom pretty much, meaning you have to scroll to see them on certain devices (my phone and laptop). Would make navigation easier and more convenient if these frequently used sections were higher up in the list.

These changes put the most frequently used items together near the top, so users won't have to read or scroll as much in order to access them.

### LINKED ISSUE #4188: Singular or plural...
Singular or plural... that is the question:

image

Customization vs Persons vs Automation vs Server Control?
Customizations vs Persons vs Automations vs Server Controls?
Customization vs Person vs Automation vs Server Control?

**Is your feature request related to a problem? Please describe.**

🤷‍♂ 

**Describe the solution you'd like**

Consistency 👍 

**Describe alternatives you've considered**

Ignoring is an option 😉

### PR DESCRIPTION
**Pull request by ludeeus**
Closes #4188 and closes #4040 

## Changes

- Reorders the sections
- Updates the base translations for some of the sections to conform to be pleural
- Rewrites the dashboard with LitElement/TypeScript

### Before

image

### After

image

### CODE REVIEW

**Comment by ludeeus** [comment]
Should "customizations" also be up there with automations, scripts, scenes?

**Comment by bramkragten** [comment]
No, I don't think so

File: src/panels/config/dashboard/ha-config-dashboard.js [hunk 1]
```diff
@@ -103,18 +103,45 @@ class HaConfigDashboard extends NavigateMixin(LocalizeMixin(PolymerElement)) {
...
+                      [[localize('ui.panel.config.scene.description')]]
+                    </div>
+                  </paper-item-body>
+                  <ha-icon-next></ha-icon-next>
+                </paper-item>
+              </a>
+            </template>
           </ha-card>
 
           <ha-config-navigation
```
Comment (bramkragten) [inline src/panels/config/dashboard/ha-config-dashboard.js]: We should makes this element accept `PAGES` as a property and use it for both of them.

**Review by bramkragten — APPROVED** [review]

File: src/panels/config/dashboard/ha-config-navigation.ts [hunk 2]
```diff
@@ -16,34 +16,20 @@ import {
...
-  { page: "zha" },
-  { page: "zwave" },
-  { page: "customize", core: true, advanced: true },
-];
-
 @customElement("ha-config-navigation")
 class HaConfigNavigation extends LitElement {
   @property() public hass!: HomeAssistant;
   @property() public showAdvanced!: boolean;
+  @property() public pages!: Array<{
```
[the remark this answers is not in the document]
Comment (ludeeus) [reply src/panels/config/dashboard/ha-config-navigation.ts]: Should I just export it where it is, or move it to data with the other?

[the remark this answers is not in the document]
Comment (bramkragten) [reply src/panels/config/dashboard/ha-config-navigation.ts]: No you can export it from `ha-config-navigation` as that's where the type belongs to.

File: src/panels/config/dashboard/ha-config-dashboard.ts [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+import "../ha-config-section";
+import "./ha-config-navigation";
+
+@customElement("ha-config-dashboard")
+class HaConfigDashboard extends LitElement {
+  @property() public hass!: HomeAssistant;
+  @property() public narrow!: boolean;
+  @property() public isWide!: boolean;
+  @property() public cloudStatus!: CloudStatusLoggedIn;
```
[the remark this answers is not in the document]
Comment (ludeeus) [reply src/panels/config/dashboard/ha-config-dashboard.ts]: Both works, I'm not logged in in my dev env.
But using `CloudStatus` will also give an "error"
image

File: src/panels/config/dashboard/ha-config-dashboard.ts [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+                <paper-item-body two-line="">
+                ${this.hass.localize("ui.panel.config.cloud.caption")}
+                  ${
+                    this.cloudStatus.logged_in
+                      ? html`
+                          <div secondary="">
+                            ${this.hass.localize(
+                              "ui.panel.config.cloud.description_login",
+                              "email",
+                              this.cloudStatus.email
```
Comment (bramkragten) [inline src/panels/config/dashboard/ha-config-dashboard.ts]:
```suggestion
                              (this.cloudStatus as CloudStatusLoggedIn).email
```

**Review by bramkragten — APPROVED** [review]
Nice! 🎉

**Pull request merged**